packages/flame_jenny/jenny/lib/src/structure/expressions/expression.dart

```diff
@@ -3,18 +3,17 @@ abstract class Expression {
 
   dynamic get value;
 
-  bool get isNumeric => false;
-  bool get isBoolean => false;
-  bool get isString => false;
+  bool get isNumeric => type == ExpressionType.numeric;
+  bool get isBoolean => type == ExpressionType.boolean;
+  bool get isString => type == ExpressionType.string;
 
   ExpressionType get type {
-    return isNumeric
-        ? ExpressionType.numeric
-        : isBoolean
-            ? ExpressionType.boolean
-            : isString
-                ? ExpressionType.string
-                : ExpressionType.unknown;
+    return switch (this) {
```

Comment:
I think this new way will be less performant if the getters will be used a lot when going through a deep tree for example, but it's way more readable. I guess it's not worth optimizing for performance here already though!